### PR TITLE
Feature/page navigation

### PR DESCRIPTION
**Pull request by kevinmpowell**
@nathanacurtis There are actually three sink pages for this one testing various ways to invoke and customize the macro. See the homepage for links to all 3 sink pages.

### CODE REVIEW

File: components/page_navigation/page_navigation.njk [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                        anchor_link_target_selector="h2, h3", 
+                        fixed_distance_from_top=0) %}
+    <nav class="esds-doc-page-navigation{{ ' ' + class if class }}" 
+        {% if items.length == 0 %}
+            data-esds-doc-anchor-link-target-selector="{{ anchor_link_target_selector }}"
+        {% endif %}
+        data-esds-doc-fixed-distance-from-top="{{ fixed_distance_from_top }}"
+        >
+        <ul class="esds-doc-page-navigation__inner">
+            {% if items.length > 0 %}
```
Comment (nathanacurtis) [inline components/page_navigation/page_navigation.njk]: Is there not support for indented subheads? It's unclear to me how this loop can generate H3s.

File: components/page_navigation/page_navigation.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
+$esds-doc-page-navigation-max-width: 240px !default;
```
Comment (nathanacurtis) [inline components/page_navigation/page_navigation.scss]: I'd like to see us start the convention of having these variables in a separate file, as we should be doing on MDS. Not in this PR, but soon.

File: components/page_navigation/page_navigation.scss [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    width: 100%;
+}
+
+.esds-doc-page-navigation__inner {
+    list-style: none;
+    margin: 0;
+    padding: $esds-doc-page-navigation-space-inset;
+}
+
+.esds-doc-page-navigation__link {
```
Comment (nathanacurtis) [inline components/page_navigation/page_navigation.scss]: I'd like to see us more directly provide default null styles via variables (but... probably not mixins?) for key properties likely to be overridden, like `font-family:`, `margin:`, and others. Shouldn't we be able to more quickly style this in another repo using variables?

File: tokens/tokens.yaml [hunk 4]
```diff
@@ -120,6 +120,12 @@ space:
     half-x: 0 0 8px
     quarter-x: 0 0 4px
 
+icon:
```
Comment (nathanacurtis) [inline tokens/tokens.yaml]: There's alot of `icons` stuff in this `page_navigation` PR. Can we more cleanly separate things in future PRs?

File: components/page_navigation/page_navigation.scss (hunk 2 above)
Comment (kevinmpowell) [reply components/page_navigation/page_navigation.scss]: 👍

File: components/page_navigation/page_navigation.njk (hunk 1 above)
Comment (kevinmpowell) [reply components/page_navigation/page_navigation.njk]: I haven't added this yet. We can add that as a non-breaking feature in a later release. Sound good?

File: tokens/tokens.yaml (hunk 4 above)
Comment (kevinmpowell) [reply tokens/tokens.yaml]: Yes, that's poor branch management on my end.

File: components/page_navigation/page_navigation.njk (hunk 1 above)
Comment (nathanacurtis) [reply components/page_navigation/page_navigation.njk]: Sure

File: components/page_navigation/page_navigation.scss (hunk 3 above)
Comment (kevinmpowell) [reply components/page_navigation/page_navigation.scss]: Yes, I'm feeling a need for that as I'm using the doc components. Especially stack spacing. I need to give this more thought.

**Pull request merged**